File: csharp/MkCert/Decrypt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Org.BouncyCastle.Crypto.Parameters;
using Org.BouncyCastle.Crypto.Digests;
using Org.BouncyCastle.Crypto.Engines;
using Org.BouncyCastle.Crypto.Encodings;
using System.Security.Cryptography;

namespace MkCert
{
    public class Decrypt
    {
        public static byte[] DecryptBytes(RsaPrivateCrtKeyParameters privateKey, byte[] cipherContent)
        {
            var rsaEngine = new Org.BouncyCastle.Crypto.Engines.RsaEngine();
            var oaepEncoding = new Org.BouncyCastle.Crypto.Encodings.OaepEncoding(rsaEngine, new Sha256Digest(), new Sha256Digest(), null); // default for MGF1-SHA256
            oaepEncoding.Init(false, privateKey);
            var inputBlockSize = oaepEncoding.GetInputBlockSize();
            using (System.IO.MemoryStream ms = new System.IO.MemoryStream())
            {

                System.IO.BinaryWriter bw = new System.IO.BinaryWriter(ms);

                for (int i = 0; i < cipherContent.Length; i += inputBlockSize)
                {
                    var ciphertext = oaepEncoding.ProcessBlock(cipherContent, i, i + inputBlockSize < cipherContent.Length ? inputBlockSize : cipherContent.Length - i);
                    bw.Write(ciphertext);
                }
                bw.Flush();
                return ms.ToArray();
            }
        }

        public static byte[] DecryptBytes(RsaPrivateCrtKeyParameters privateKey, string base64EncodedCipherContent)
        {
            var cipherContent = Org.BouncyCastle.Utilities.Encoders.Base64.Decode(base64EncodedCipherContent);
            return DecryptBytes(privateKey, cipherContent);
        }

        public static byte[] DecryptWithRSACng(RsaPrivateCrtKeyParameters privateKey, byte[] cipherContent)
        {
            using (RSA rsa = RSA.Create())
            {
                rsa.ImportParameters(new RSAParameters()
                {
                    D = privateKey.Exponent.ToByteArray(),
                    DP = privateKey.DP.ToByteArray(),
                    DQ = privateKey.DQ.ToByteArray(),
                    Modulus = privateKey.Modulus.ToByteArray(),
                    P = privateKey.P.ToByteArray(),
                    Q = privateKey.Q.ToByteArray(),
                    Exponent = privateKey.PublicExponent.ToByteArray(),
                    InverseQ = privateKey.QInv.ToByteArray()
                });

                return rsa.Decrypt(cipherContent, RSAEncryptionPadding.OaepSHA256);
            }
        }
    }
}

File: java/src/main/java/Decrypt.java
import org.bouncycastle.crypto.params.RSAPrivateCrtKeyParameters;

/**
 * Created by beeven on 12/20/2016.
 */

import org.bouncycastle.crypto.InvalidCipherTextException;
import org.bouncycastle.crypto.digests.SHA256Digest;
import org.bouncycastle.crypto.encodings.OAEPEncoding;
import org.bouncycastle.crypto.engines.RSAEngine;
import org.bouncycastle.crypto.params.RSAPrivateCrtKeyParameters;

import java.io.ByteArrayOutputStream;
import java.io.IOException;

public class Decrypt {

    public static byte[] DecryptBytes(RSAPrivateCrtKeyParameters privateKeyParam, byte[] cipherContent) throws InvalidCipherTextException, IOException {
        RSAEngine rsaEngine = new RSAEngine();
        OAEPEncoding oaepEncoding = new OAEPEncoding(rsaEngine, new SHA256Digest(), new SHA256Digest(), null);
        oaepEncoding.init(false, privateKeyParam);
        int inputBlockSize = oaepEncoding.getInputBlockSize();
        ByteArrayOutputStream bo = new ByteArrayOutputStream();
        for(int i=0;i< cipherContent.length; i+=inputBlockSize) {
            byte[] decryptedBytes = oaepEncoding.processBlock(cipherContent, i, i + inputBlockSize < cipherContent.length ? inputBlockSize: cipherContent.length - i);
            bo.write(decryptedBytes);
        }
        bo.flush();
        return bo.toByteArray();
    }

    public static byte[] DecryptBytes(RSAPrivateCrtKeyParameters privateKeyParam, String base64EncodedCipherContent) throws IOException, InvalidCipherTextException {
        byte[] cipherContent = org.bouncycastle.util.encoders.Base64.decode(base64EncodedCipherContent);
        return DecryptBytes(privateKeyParam, cipherContent);
    }
}
